components/support/ktx/src/main/java/mozilla/components/support/ktx/android/net/Uri.kt

```diff
@@ -25,3 +25,27 @@ val Uri.hostWithoutCommonPrefixes: String?
  */
 val Uri.isHttpOrHttps: Boolean
     get() = scheme == "http" || scheme == "https"
+
+/**
+ * Checks that the given URL is in scope of the web app specified by [trustedScopes].
+ *
+ * https://www.w3.org/TR/appmanifest/#dfn-within-scope
+ */
+fun Uri.isInScope(trustedScopes: Iterable<Uri>): Boolean {
+    val path = path.orEmpty()
+    return trustedScopes.any { scope ->
+        sameOriginAs(scope) && path.startsWith(scope.path.orEmpty())
+    }
+}
+
+/**
+ * Checks that Uri has the same host as [other].
+ */
+fun Uri.sameHostAs(other: Uri) = scheme == other.scheme && host == other.host
```

Comment:
I made these methods public because there are similar checks all over Fenix and AC. I kept the refactoring limited to PWA-related code, but other checks exist with the same names. I believe same host code in the engine used to be called in multiple locations.